local_pathfinding/node_navigate.py

```diff
@@ -134,17 +131,35 @@ def get_desired_heading(self) -> float:
             float: The desired heading if all subscribers are active, else a number that violates
                 the heading convention.
         """
+        _, planner = self.get_params()
+
         if not self._all_subs_active():
             self._log_inactive_subs_warning()
             return -1.0
 
         self.local_path.update_if_needed(
-            self.gps, self.ais_ships, self.global_path, self.filtered_wind_sensor
+            self.gps, self.ais_ships, self.global_path, self.filtered_wind_sensor, planner
         )
 
         # TODO: create function to compute the heading from current position to next local waypoint
         return 0.0
 
+    def get_params(self):
```

Comment:
Let's do something a bit fancier here. Only when the parameter has changed:
- Save to instance variables (so don't have to return anything)
- Debug log
- [Only when `pub_period_sec` changes] Update timers
    - So you probably want to move the timer initializations to inside this function

After you have made these changes you should verify them by trying to change them while local pathfinding is running:
- In one terminal: `ros2 launch local_pathfinding main_launch.py`
- In another terminal: `ros2 set param navigate_main <param> <value>`